website/docs/r/kinesis_firehose_delivery_stream.html.markdown

```diff
@@ -53,6 +53,88 @@ resource "aws_kinesis_firehose_delivery_stream" "test_stream" {
 }
 ```
 
+### Extended S3 Destination
+
+```hcl
+resource "aws_s3_bucket" "bucket" {
+  bucket = "tf-test-bucket"
+  acl    = "private"
+}
+
+resource "aws_iam_role" "firehose_role" {
+  name = "firehose_test_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "firehose.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role" "lambda_iam" {
+  name = "lambda_iam"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_function" "lambda_processor" {
+  filename = "lambda.zip"
+  function_name = "firehose_lambda_processor"
+  role = "${aws_iam_role.lambda_iam.arn}"
+  handler = "exports.handler"
+  runtime = "nodejs4.3"
+}
+
+resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
```

Comment:
Because this example requires a lot of boiler plate to actually execute, can we move the `aws_kinesis_firehose_delivery_stream` to the top? That way users immediately see the `aws_kinesis_firehose_delivery_stream` and it's `extended_s3_configuration`, and they can continue reading into the lambda and IAM role bits if then need/want.